fvm/errors/errors.go

```diff
@@ -213,3 +214,29 @@ func NewEventEncodingError(err error) CodedError {
 		ErrCodeEventEncodingError,
 		"error while encoding emitted event: %w ", err)
 }
+
+// EVMError needs to satisfy the user error interface
+// in order for Cadence to correctly handle the error
+var _ errors.UserError = &(EVMError{})
+
+type EVMError struct {
+	CodedError
+}
+
+func (e EVMError) IsUserError() {}
```

Comment:
So is this the missing part ?

Reply:
Basically, this makes it conform to the type of Cadence User Error which doesn't contain the stack and is handled gracefully. 
Now we could maybe even go further and adding an EVM error type in Cadence handling which I think could be even nicer way to go, but this solves it for now and we can add it to the todo if you think so?